src/expr.rs

```diff
@@ -1779,7 +1779,10 @@ fn rewrite_call_args(context: &RewriteContext,
     };
 
     match write_list(&item_vec, &fmt) {
-        Some(ref s) if !s.contains('\n') && s.len() > one_line_width => {
+        // If arguments do not fit in a single line and do not contain newline,
+        // try to put it on the next line.
+        Some(ref s) if context.config.fn_call_style == IndentStyle::Block &&
+                       (!s.contains('\n') && s.len() > one_line_width) => {
```

Comment:
Thanks for comments! I have 2 questions.
1. Why do we have to distinguish macro and funcation call here? We have the same bug with function call as well.
2. With block mode, I get the following output:
```rust
fn a() {
    panic(
        "this is a long string that goes past the maximum line length causing rustfmt to insert a comma here:",
    );
}
```
That is, rustfmt tries multiline because args don't fit in a single line and hence needs trailing comma, right?